test/env/lib.rs

```diff
@@ -147,8 +148,27 @@ impl Transfer {
     }
 }
 
+#[async_trait(?Send)]
+pub trait LedgerEnv {
+    fn fork(&self) -> Self;
+    async fn transfer(&self, arg: Transfer) -> anyhow::Result<Result<Nat, TransferError>>;
+    fn principal(&self) -> Principal;
+    async fn balance_of(&self, account: impl Into<Account>) -> anyhow::Result<Nat>;
+    async fn supported_standards(&self) -> anyhow::Result<Vec<SupportedStandard>>;
+    async fn metadata(&self) -> anyhow::Result<Vec<(String, Value)>>;
+    async fn minting_account(&self) -> anyhow::Result<Option<Account>>;
+    async fn token_name(&self) -> anyhow::Result<String>;
+    async fn token_symbol(&self) -> anyhow::Result<String>;
+    async fn token_decimals(&self) -> anyhow::Result<u8>;
+    async fn transfer_fee(&self) -> anyhow::Result<Nat>;
```

Comment:
It's usually beneficial to keep traits as small as possible.

Note that the `query` endpoint is enough to implement many other methods.
I suggest we keep only `query` and `update` (it's not in the interface yet, but could be) methods in the interface and implement everything else as either free functions or extension traits.

Reply:
I split LedgerEnv into two traits (LedgerEnv,LedgerEndpoint):
I would argue that `principal()` and `fork()` also need to be in the LedgerEnv trait as they are used in the tests multiple times and in the tests we only specify the traits not the struct implementation. Free-floating functions would only be possible for the `new(...)` operator. 

I also added the `update(...)` function to LedgerEnv